zap/src/main/java/org/zaproxy/zap/extension/ascan/CustomScanDialog.java

```diff
@@ -902,6 +922,13 @@ public String validateFields() {
             return Constant.messages.getString("ascan.custom.notSafe.error");
         }
 
+        if (this.filterPanel != null) {
```

Comment:
We really should not need to check for null (we don't do the check in the save method so it would NPE in the end). Worth creating it in the constructor to avoid these.

Reply:
actually it is tough reason being filterpanel instance requires Target and target is provided in init method. 
if for removing a null check, we are initialising panel with null target, i think it is not required. o

Reply:
The panel does not need/require the target to be created/initialised, so it can be removed. In `init` method it just needs to call `resetFilterPanel` with the target.